client/platform/web-girder/views/Export.vue

```diff
@@ -14,32 +23,68 @@ export default {
     },
   },
 
-  data() {
-    return {
-      menuOpen: false,
-      excludeFiltered: false,
-      activator: 0,
-    };
-  },
+  setup(props) {
+    const menuOpen = ref(false);
+    const excludeFiltered = ref(false);
+    const exportUrls = ref(null as null | ExportUrlsResponse);
+    const savePrompt = ref(false);
+    const currentSaveUrl = ref('');
+    let save = () => Promise.resolve();
+    let pendingSaveCount = ref(0);
+
+    try {
+      save = useHandler().save;
+      pendingSaveCount = usePendingSaveCount();
+    } catch (err) {
+      /**
+       * Error indicates that this export dialog was opened in a place
+       * outside the viewer where pending changes aren't possible.
+       */
+    }
 
-  asyncComputed: {
-    async exportUrls() {
-      if (this.menuOpen) {
-        return getExportUrls(this.datasetId, this.excludeFiltered);
+    async function doExport({ forceSave = false, url }: { url?: string; forceSave?: boolean }) {
+      if (pendingSaveCount.value > 0 && forceSave) {
+        await save();
```

Comment:
On an error the `save()` function with automatically show an error prompt and cancel the saving of the file.  The user hits okay and then it moves along in the function.  I think we need to modify the `save()` function to respond back if it is successful or not here with a boolean value or perhaps the error.  When it is not successful you want to return and prevent the rest of this `doExport` function from executing.

Easy example is to log in as another user and attempt to edit public dataset of a different user.  On export without saving it will pop open the export prompt, you click on save it will pop open the error prompt indicating you don't have permissions.  Clicking OK on that will cause you to fall through the rest of this function where it attempts to change the URL so another prompt comes up asking if you want to leave the page.  If you click yes it will download the detections without your save in there.

Reply:
More of an error handling issue, I think.  I resolved this by re-throwing the error after the prompt.  I think it's more useful to have `await save()` live in try/catch and expose the error object that way than to have to pass back true/false and obscure the error from downstream handlers.

This case does get you "stuck" in a situation where there are pending changes you can't save so you can't export until you back out (and discard) and come back again, but I think that problem is out of scope for this issue.  Adding some discard-changes feature would resolve this.

What do you think?

Reply:
Yeah I like this, the stuck-ness I think is more a result of "not informing the user that their changes can't be saved until they attempt to save them" which is a different issue.